components/Tel.Egram.Components.Application/ApplicationController.cs

```diff
@@ -28,18 +29,50 @@ public class ApplicationController : BaseController<MainWindowModel>
             IAuthenticator authenticator,
             IApplicationPopupController applicationPopupController,
             IActivator<AuthenticationModel> authenticationActivator,
-            IActivator<WorkspaceModel> workspaceActivator)
+            IActivator<WorkspaceModel> workspaceActivator,
+            IAgent agent)
         {
             _authenticationActivator = authenticationActivator;
             _workspaceActivator = workspaceActivator;
-            
+
+            BindConnection(agent)
+                .DisposeWith(this);
+
             BindAuthenticator(authenticator)
                 .DisposeWith(this);
 
             BindPopup(applicationPopupController)
                 .DisposeWith(this);
         }
 
+        private IDisposable BindConnection(IAgent agent)
+        {
+            return agent
+                .ObserveConnectionState()
+                .ObserveOn(RxApp.MainThreadScheduler)
+                .Subscribe(state =>
+                {
+                    switch (state)
+                    {
+                        case TdApi.ConnectionState.ConnectionStateConnecting _:
+                            UpdateConnectionState(0);
```

Comment:
Personally I would create an enum representing connection state instead of using numbers.